File: admin-ui/lib/api/types/UserDto.ts
import {TelegramUserDto} from "@/lib/api/types/telegram/TelegramUserDto";

export type UserDto = {
    id: number;
    telegramUser: TelegramUserDto;
    role: UserDto.Role;
    smallPhotoUrl: string | null;
    bigPhotoUrl: string | null;
};

export namespace UserDto {
    export type Role = "ADMIN";
}

File: src/main/kotlin/space/davids_digital/kiri/rest/dto/UserDto.kt
package space.davids_digital.kiri.rest.dto

import space.davids_digital.kiri.rest.dto.telegram.TelegramUserDto

data class UserDto (
    val id: Long,
    val telegramUser: TelegramUserDto,
    val role: Role,
    val smallPhotoUrl: String?,
    val bigPhotoUrl: String?
) {
    enum class Role {
        OWNER,
        ADMIN
    }
}